app/controllers/feedback_controller.rb

```diff
@@ -0,0 +1,58 @@
+class FeedbackController < ApplicationController
+  before_action :update_return_path
+
+  def new
+    feedback
+  end
+
+  def create
+    feedback.update(feedback_params)
+    FeedbackMailer.notify(feedback).deliver_later if feedback_submitted?
+
+    redirect_to feedback
+  end
+
+  def show
+    @feedback = Feedback.find(params[:id])
+  end
+
+  private
+
+  def feedback_params
+    params.require(:feedback).permit(
+      :done_all_needed, :satisfaction, :improvement_suggestion
+    )
+  end
+
+  def feedback
+    @feedback ||= Feedback.new(
+      source: source,
+      os: browser.platform.name,
+      browser: browser.name,
+      browser_version: browser.full_version
+    )
+  end
```

Comment:
I don't think we are capturing the actual page they were on when they clicked the feedback link (although I see from the ticket that isn't down as a requirement, but it is something we discussed).